src/pip/_internal/download.py

```diff
@@ -60,15 +62,38 @@
 
 if MYPY_CHECK_RUNNING:
     from typing import (
-        Optional, Tuple, Dict, IO, Text, Union
+        Callable, Dict, List, IO, Optional, Text, Tuple, Union
     )
     from optparse import Values
+
+    from mypy_extensions import TypedDict
```

Comment:
Just a thought -- does mypy care whether mypy_extensions is installed when it's analyzing code or not?

Reply:
It is provided transitively through mypy itself, so if mypy is in the environment then `mypy_extensions` is in the environment. I can add it in the mypy requirements file explicitly.

Reply:
On second thought since we pin the `mypy` version, I think I will leave it as-is. Better to get the automatic update when we update `mypy` with the very minor risk that it gets removed later on than to have to update two pinned dependencies, IMO.